cmd/ecs-deploy/task_def.go

```diff
@@ -114,36 +116,89 @@ const (
 
 // ECRImage represents an ECR Image tag broken into its constituent parts
 type ECRImage struct {
-	AWSRegion      string
-	imageURI       string
-	ImageTag       string
-	RegistryID     string
-	RepositoryURI  string
-	RepositoryName string
+	AWSRegion        string
+	Digest           string
+	ImageURIByDigest *string
+	ImageURIByTag    *string
+	RegistryID       string
+	RepositoryURI    string
+	RepositoryName   string
+	Tag              string
 }
 
 // NewECRImage returns a new ECR Image object
-func NewECRImage(imageURI string) (*ECRImage, error) {
-	imageParts := strings.Split(imageURI, ":")
+func NewECRImage(imageURI string, serviceECR *ecr.ECR) (*ECRImage, error) {
```

Comment:
You can remove `serviceECR` here:

```suggestion
func NewECRImage(imageURI string) (*ECRImage, error) {
```